rust/src/nfs/nfs2_records.rs

```diff
@@ -96,3 +96,211 @@ pub fn parse_nfs2_attribs(i: &[u8]) -> IResult<&[u8], Nfs2Attributes> {
     let attrs = Nfs2Attributes { atype, asize };
     Ok((i, attrs))
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::nfs::nfs2_records::*;
+    use nom7::bytes::streaming::take;
+    use nom7::IResult;
+
+    #[test]
+    fn test_nfs2_handle() {
+
+        // file_handle bytes
+        let buf: &[u8] = &[
+            0x00, 0x10, 0x10, 0x85, 0x00, 0x00, 0x03, 0xe7,
+            0x00, 0x0a, 0x00, 0x00, 0x00, 0x00, 0xb2, 0x5a,
+            0x00, 0x00, 0x00, 0x29, 0x00, 0x0a, 0x00, 0x00,
+            0x00, 0x00, 0xb2, 0x5a, 0x00, 0x00, 0x00, 0x29
+        ];
+
+        let result = parse_nfs2_handle(buf);
+        match result {
+            Ok((r, res)) => {
+                assert_eq!(r.len(), 0);
+                assert_eq!(res.value, buf);
+            }
+            Err(error) => { panic!("Parsing nfs2_handle failed: {:?}", error); }
+        }
+    }
+
+    #[test]
+    fn test_nfs2_request_lookup() {
+
+        // packet_bytes
+        let buf: &[u8] = &[
+
+        // [file_handle]
+            0x00, 0x10, 0x10, 0x85, 0x00, 0x00, 0x03, 0xe7,
+            0x00, 0x0a, 0x00, 0x00, 0x00, 0x00, 0xb2, 0x5a,
+            0x00, 0x00, 0x00, 0x29, 0x00, 0x0a, 0x00, 0x00,
+            0x00, 0x00, 0xb2, 0x5a, 0x00, 0x00, 0x00, 0x29,
+
+        // [name]
+        // name_len
+            0x00, 0x00, 0x00, 0x02,
+
+        // name_contents: (am)
+            0x61, 0x6d,
+
+        // _name_padding
+            0x00, 0x00,
+        ];
+
+        let (_, handle) = parse_nfs2_handle(buf).expect("Parsing nfs2_handle failed!");
+
+        let result = parse_nfs2_request_lookup(buf);
+        match result {
+            Ok((r, request)) => {
+                assert_eq!(r.len(), 0);
+                assert_eq!(request.handle, handle);
+                assert_eq!(request.name_vec, b"am".to_vec());
+            }
+            Err(error) => { panic!("Parsing failed: {:?}", error); }
+        }
+    }
+
+    #[test]
+    fn test_nfs2_request_read() {
+
+        // packet_bytes
+        let buf: &[u8] = &[
+
+        // [file_handle]
+            0x00, 0x10, 0x10, 0x85, 0x00, 0x00, 0x03, 0xe7,
+            0x00, 0x0a, 0x00, 0x00, 0x00, 0x00, 0xb2, 0x5d,
+            0x00, 0x00, 0x00, 0x2a, 0x00, 0x0a, 0x00, 0x00,
+            0x00, 0x00, 0xb2, 0x5a, 0x00, 0x00, 0x00, 0x29,
+
+        // offset
+            0x00, 0x00, 0x00, 0x00,
+        // count
+            0x00, 0x00, 0x20, 0x00,
+        // total count
+            0x00, 0x00, 0x20, 0x00
+        ];
+
+        let (_, handle) = parse_nfs2_handle(buf).expect("Parsing nfs2_handle failed!");
+
+        let result = parse_nfs2_request_read(buf);
+        match result {
+            Ok((r, request)) => {
+                assert_eq!(r.len(), 4);
+                assert_eq!(request.handle, handle);
+                assert_eq!(request.offset, 0);
+            }
+            Err(error) => { panic!("Parsing failed: {:?}", error); }
+        }
+    }
+
+    #[test]
+    fn test_nfs2_reply_read() {
+
```

Comment:
We could perhaps add the pcap data alongside the hex bytes and the source here. Excellent example: https://github.com/OISF/suricata/blob/master/rust/src/dns/dns.rs#L1013

Reply:
this seems like testing against the buffer as a whole. should I discard specifying the fields?
I understand the tests are too verbose, trying to figure out how to make it compact and clear at the same time

I found this one with comments in the middle.
https://github.com/OISF/suricata/blob/4972021cd4c362c6f02fce6c34377013a463eab0/rust/src/dns/dns.rs#L1234